lib/editing/editor_codemirror.dart

```diff
@@ -240,6 +248,22 @@ class _CodeMirrorDocument extends Document {
 ////      cm.setGutterMarker(an.line - 1, _gutterId,
 ////          _makeMarker(an.type, an.message, an.start, an.end));
     }
+    var path = html.querySelector("#runbutton path");
```

Comment:
Could you move the run button updating code into it's own method? That'll make sure this `setAnnotations` method doesn't get too long. Some thing like `_updateRunButton({bool hasError: false, bool hasWarning: false})`.